src/components/geolocated.js

```diff
@@ -64,17 +65,25 @@ const geolocated = ({
         }
 
         getLocation() {
-            if (!geolocationProvider || !geolocationProvider.getCurrentPosition) {
+            if (!geolocationProvider || !geolocationProvider.getCurrentPosition || !geolocationProvider.watchPosition) {
                 throw new Error('The provided geolocation provider is invalid');
             }
 
+            const funcPosition = (watchPosition
+              ? geolocationProvider.watchPosition
+              : geolocationProvider.getCurrentPosition).bind(geolocationProvider);
```

Comment:
looks weird, but `getCurrentPosition/watchPosition` require a geolocation context to work.